tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -199,7 +199,7 @@ private SpanContext(TraceId traceId, string serviceName)
         /// <summary>
         /// Gets the span id.
         /// </summary>
-        public ulong SpanId { get; }
+        public ulong SpanId { get; internal set; }
```

Comment:
This feels iffy.

Reply:
So part of the [W3C Phase 3 ](https://docs.google.com/document/d/1Zc7uAQTJ2G5vHEbmx8k4vMnYdQ-BnqZxewXk0ohgd3g/edit#heading=h.79mohmi01gqf)is the future deprecation of Datadog headers, so when we find conflicting headers we prioritize W3C span id and need to update them, in the latest commit I do so like:
https://github.com/DataDog/dd-trace-dotnet/pull/5721/commits/fdd4fd58fd0d831224140cc6ad68670c261593ab#diff-3f905f6ce54156702b562c581f1ff7866c41fbafb6b29b79c6c2411c79c19e55R187-R188

```
localSpanContext.SpanId = spanContext.SpanId;
localSpanContext.RawSpanId = spanContext.RawSpanId;
```
Do you have any other prefer methods, I Googled the `internal` set and found no complaints so far.

Reply:
Oh, this is `SpanContext`. I thought this was `Span`. 😅 

Still, it would've been nice to mutate things as little as possible. Maybe determine the final span id first before creating the `SpanContext`, but that's a larger refactor.

Thanks for clarifying.